content/References/dRAIDprimer.md

```diff
@@ -0,0 +1,122 @@
+
+Current investigations between dRAID and RAIDZ vdev layouts find that RAIDZ layouts store data more efficiently in all general use case scenarios, and especially in scenarios where small blocks of data are being stored.
+{{< /hint >}}
+
+These images demonstrate the differences between dRAID and raidz layouts in OpenZFS:
+
+{{< columns >}}
+{{< trueimage src="/images/Reference/raidzdRAIDSimplified.png" alt="Simplified dRAID and raidz characterizations" id="Simplified raidz and dRAID characterization" >}}
+<--->
+{{< trueimage src="/images/Reference/dRAIDandraidz.png" alt="dRAID and raidz characterizations" id="dRAID Characterization" >}}
+<p style="text-align: center;"><small>Thanks to Zhi (George) Qiao, Song Fu, Hsing-bung Chen, and Bradley Wade Settlemyer.</small></p>
+{{< /columns >}}
+
+## Concepts
+
+**Healing resilver**
+
+The traditional ZFS resilver.
+The entire block tree is scanned and traversed.
+Checksums are verified during the repair process.
+This can be a slow process as it results in a largely random workload that is not ideal for performance.
+In a RAIDZ deployment this also puts extra strain on the remaining disks in the vdev, as they are all being read from simultaneously.
+
+**Permutation maps**
+
+dRAID uses an array of predetermined permutation maps to determine where data, parity, and spare capacity reside across the pool.
+This ensures that during resilvers, all IO (reads and writes) distribute evenly across all disks, reducing the load on any one disk.
+
+Because a permutation map automatically selects during pool creation, **distributed spares cannot be added after pool creation**.
+If adding spares after pool creation is a critical requirement, create the pool using a raidz layout.
+
+**Sequential resilver**
+
+This is a new resilver type used by dRAID where blocks are read and restored sequentially.
+This greatly increases resilver operation speed as it does not require block tree traversal.
+
+Sequential resilver is made possible by the use of fixed stripe width in dRAID.
+During a sequential resilver data is read from all pool members.
+
+Checksums are not validated during a sequential resilver.
+However, a scrub begins after the sequential resilver finishes and verifies data checksums.
+
+**Fixed stripe width**
+
+Stripe width is fixed in dRAID, with zeroes added as padding.
+This can have a significant effect on usable capacity, depending on the data stored.
+
+In a redundancy group of eight data disks using 4k sector disks, the minimum allocation size is **32k**.
+Any files smaller than 32k still fill an entire stripe, with zeroes appended to the write to fill the entire stripe.
+This greatly reduces the pool usable capacity when the pool stores large numbers of small files.
+
+dRAID datasets benefit greatly from larger **record sizes**.
+**128k** is recommended as a minimum value for datasets and **64k** to **128k** for zvols.
```

Comment:
That will be the very minimum. Wonder if we should mention reasonable defaults, like 1M for datasets to achieve better compression. Might want to consult with mav.

Reply:
I spoke with Mav and updated this content with some more detail based on expected I/O use for datasets/zvols. Let me know if that is ok or if we're getting too "in the weeds" here. Thanks!